### ISSUE TITLE
Assertion failure: enumBuilder.m_typeBuilder == m_TypeBuilderDict[name] at ModuleBuilder.cs:1228

### ISSUE THREAD

**Issue by RussKeldorph** [dotnet/coreclr#7669]
Repro steps:

```
Build coreclr Windows x64 checked (build.cmd x64 checked skiptests)
Run corefx tests with above
    build.cmd -Release -- /p:BUILDTOOLS_OVERRIDE_RUNTIME=<coreclr_root>\bin\Product\Windows_NT.x64.Checked
```

Failing test assembly: System.Reflection.Emit.Tests.dll

coreclr commit: bc6bd52c44209647bc2541dad55a05cb2058f76f
corefx commit: dotnet/corefx@9b270c7b06751ab24d98d1404e4fd671c63aeb9e

**Comment by RussKeldorph** [comment]
@gkhanna79

**Comment by gkhanna79** [comment]
@RussKeldorph What happens during a release run of the product? Does the test pass?

@karelz @weshaggard Can you please have someone look into this asap?

**Comment by RussKeldorph** [comment]
[This](https://ci.dot.net/job/dotnet_corefx/job/release_1.1.0/) is passing, so I assume the test passes.  I can't say whether the assertion indicates a product bug, but the popup is problematic for coreclr Checked builds, which affects our ability to find other bugs.

**Comment by karelz** [comment]
@AtsushiKan owns Reflection ...

**Comment by AtsushiKan** [comment]
Fixed issue title to indicate the assert that's actually firing.

Assert is triggered by this recently added test code:

    new EnumBuilder("a\0b\0c\0", ...)

**Comment by hughbe** [comment]
Yeah I added this - it passed on my PC and the CI. Odd that this fails, considering that TypeBuilder has had a similar test for a while

**Comment by pgavlin** [comment]
@AtsushiKan: are you looking into this one? It's currently largely blocking the investigation into #8158.

**Comment by karelz** [comment]
Didn't we decide that these are bad tests? I think we just need to delete them.

**Comment by hughbe** [comment]
@karelz how come

**Comment by karelz** [comment]
@hughbe internal emails from 1.1 shipping when it hit JIT team hard, sorry for leaving out context. It seems to be kicking us again. Previously the easiest thing was just to disable the tests temporarily.

Re-reading the old emails again, @AtsushiKan agreed this is a product bug:
> It is assert from `Contract.Assert(name == enumBuilder.FullName);`
> which gets stored in the EnumBuilder by C#, and read back in `enumBuilder.FullName` by C++ and… need I say more?
> So it’s a product bug, and it’s not really right to blame the assert

**Comment by RussKeldorph** [comment]
@karelz @AtsushiKan Can you just disable the assertion for now to unblock #8158 investigation?

**Comment by karelz** [comment]
@RussKeldorph please check with @AtsushiKan decide what is the best course here now.

**Comment by AtsushiKan** [comment]
I doubt we'll ever fix this corner case so I have no objection to removing the assert.

**Comment by AtsushiKan** [comment]
Unsubscribing from thread as I'll be on vacation for the rest of the year.

**Comment by gkhanna79** [comment]
What are the next steps for this as it blocks https://github.com/dotnet/coreclr/issues/8158?

**Comment by karelz** [comment]
Do we want to port it to 1.1 to reduce the test failure noise there?

**Comment by RussKeldorph** [comment]
@karelz I believe the 1.1 fix (dotnet/corefx#13239) was to disable the provoking test in corefx.  Are there still known failures in 1.1 due to this issue?

**Comment by karelz** [comment]
Ah, I forgot. Thanks for pointing it out. We're covered on 1.1.
